activesupport/lib/active_support/logger.rb

```diff
@@ -46,6 +46,31 @@ def initialize(*args)
       @formatter = SimpleFormatter.new
     end
 
+    def level
+      # The log level should initially be setup in the main thread during startup,
+      # which means that if our current thread doesn't have a local level set, then
+      # we attempt to use the main thread's level, falling back to the class default
+      # otherwise.
+      Thread.current[:logger_level] || Thread.main[:logger_level] || super
+    end
+
+    def level=(level)
+      Thread.current[:logger_level] = level
+    end
+
+    def add(severity, message = nil, progname = nil, &block)
+      return true if @logdev.nil? || (severity || UNKNOWN) < level
```

Comment:
Why are you checking `@logdev.nil?`? Isn't it an i-var Rails has no control over?